src/olympia/search/filters.py

```diff
@@ -427,59 +427,105 @@ def primary_should_rules(self, search_query, analyzer):
 
         return should
 
-    def secondary_should_rules(self, search_query, analyzer):
+    def secondary_should_rules(
+            self, search_query, analyzer, rescore_mode=False):
         """Return "secondary" should rules for the query.
 
         These are the ones using the weakest boosts, they are applied to fields
         containing more text: description & summary.
 
         Applied rules:
 
-        * Look for phrase matches inside the summary (boost=3.0)
-        * Look for phrase matches inside the description (boost=2.0).
+        * Look for matches inside the summary (boost=3.0)
+        * Look for matches inside the description (boost=2.0).
 
         If we're using a supported language, both rules are done through a
         multi_match that considers both the default locale translation
         (using snowball analyzer) and the translation in the current language
         (using language-specific analyzer). If we're not using a supported
         language then only the first part is applied.
+
+        If rescore_mode is True, the match applied are match_phrase queries
+        with a slop of 5 instead of a regular match. As those are more
+        expensive they are only done in the 'rescore' part of the query.
         """
+        if rescore_mode is False:
+            query_class = query.Match
+            query_kwargs = {
+                'operator': 'and',
+            }
+            query_class_name = 'Match'
+            multi_match_kwargs = {
+                'operator': 'and',
+            }
+        else:
+            query_class = query.MatchPhrase
+            query_kwargs = {
+                'slop': 10,
+            }
+            query_class_name = 'MatchPhrase'
+            multi_match_kwargs = {
+                'slop': 10,
+                'type': 'phrase',
+            }
+
         if analyzer:
             summary_query_name = (
-                'MultiMatch(MatchPhrase(summary),'
-                'MatchPhrase(summary_l10n_%s))' % analyzer)
+                'MultiMatch(%s(summary),%s(summary_l10n_%s))' % (
+                    query_class_name, query_class_name, analyzer))
             description_query_name = (
-                'MultiMatch(MatchPhrase(description),'
-                'MatchPhrase(description_l10n_%s))' % analyzer)
+                'MultiMatch(%s(description),%s(description_l10n_%s))' % (
+                    query_class_name, query_class_name, analyzer))
             should = [
+                # Note the presence of operator='and', which may seem weird
```

Comment:
Ah yeah, I refactored that to `multi_match_kwargs` because that's an invalid keyword for when we do phrase matches in the refactor phase. It is present only when we do regular matches. I'll reword the comment to make it clearer.

Reply:
https://github.com/mozilla/addons-server/pull/9985/commits/10a382a5737ec47adf5a49ff6977a941f8c9f05e rewords the comment